examples/src/test/java/io/numaproj/numaflow/examples/map/flatmap/FlatMapFunctionTest.java

```diff
@@ -0,0 +1,64 @@
+package io.numaproj.numaflow.examples.map.flatmap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Iterator;
+
+import io.numaproj.numaflow.mapper.Datum;
+import io.numaproj.numaflow.mapper.Message;
+import io.numaproj.numaflow.mapper.MessageList;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+
+@Slf4j
+public class FlatMapFunctionTest {
+
+    @Test
+    public void testCommaSeparatedString() {
+        Datum datum = mock(Datum.class);
+        when(datum.getValue()).thenReturn("apple,banana,carrot".getBytes());
+
+        FlatMapFunction flatMapFunction = new FlatMapFunction();
+        MessageList result = flatMapFunction.processMessage(new String[]{}, datum);
+
+        Iterator<Message> iterator = result.getMessages().iterator();
```

Comment:
maybe instead of needing to use the iterator here, there is a method to get an array. Makes it a bit easier to use. or you could get a stream.

Reply:
We can provide a utility class to help transform the Iterator to Array/Stream, similar to the concrete Datum class below. cc: @yhl25

Reply:
we can expose another function `getMessagesList()` which returns the list of messages.